File: 2000 98 & ME/QuickStart/howto/samples/Services/Installers/customaction/vb/custominstaller.vb
Imports System
Imports System.Collections
Imports System.Configuration.Install

public class CustomInstaller: Inherits Installer

	public overrides sub Install(state As IDictionary)
		Context.LogMessage("Installing")
		MyBase.Install(state)
		Console.WriteLine("Installing: Any code could be executed here!")
	end sub

	public overrides sub Uninstall(state As IDictionary)
		Context.LogMessage("Uninstalling")
		Console.WriteLine("Uninstalling: Any code could be executed here!")
		MyBase.Uninstall(state)
	end sub
end class

File: 2000 98 & ME/QuickStart/howto/samples/Services/Installers/customaction/custominstaller.cs
using System;
using System.Collections;
using System.Configuration.Install;

public class CustomInstaller: Installer{

	public override void Install(IDictionary state) {
		Context.LogMessage("Installing");
		base.Install(state);
		Console.WriteLine("Installing: Any code could be executed here!");
	}

	public override void Uninstall(IDictionary state) {
		Context.LogMessage("Uninstalling");
		Console.WriteLine("Uninstalling: Any code could be executed here!");
		base.Uninstall(state);
	}
}
